### PR TITLE
Prevent top-level calls to `requestAnimationFrame`

### PR DESCRIPTION
**Pull request by bartlomiejbloniarz**
## Summary

Currently there are instances where `scheduleOnUiImmediately` is called in the top level to create global structures which are not needed on web. This results in a call to `requestAnimationFrame`, which is unavailable during static website generation. We already have a mock for `raf` which prevents the error, but in this PR I changed the logic so the unnecessary calls to `raf` don't happen on web.

### CODE REVIEW

File: src/reanimated2/layoutReanimation/sharedTransitions/ProgressTransitionManager.ts [hunk 1]
```diff
@@ -203,10 +204,25 @@ function createProgressTransitionRegister() {
...
 }
 
-runOnUIImmediately(() => {
-  'worklet';
-  global.ProgressTransitionRegister = createProgressTransitionRegister();
-})();
+if (shouldBeUseWeb()) {
+  const throwError = () => {
+    throw new Error(
+      '[Reanimated] ProgressTransitionRegister should not be used on web.'
```
Comment (tjzel) [inline src/reanimated2/layoutReanimation/sharedTransitions/ProgressTransitionManager.ts:210]:
```suggestion
      '[Reanimated] `ProgressTransitionRegister` is not available on non-native platform.'
```

**Review by tjzel — CHANGES_REQUESTED** [review]

File: src/reanimated2/UpdateProps.ts [hunk 2]
```diff
@@ -117,14 +117,19 @@ const createUpdatePropsManager = global._IS_FABRIC
     };
 
 if (shouldBeUseWeb()) {
-  const throwError = () => {
-    throw new Error(
-      '[Reanimated] UpdatePropsManager should not be used on web.'
-    );
+  const maybeThrowError = () => {
+    if (!isJest()) {
```
Comment (tjzel) [inline src/reanimated2/UpdateProps.ts:121]: Please add a comment about why do we have to if `jest` out here.

File: src/reanimated2/layoutReanimation/sharedTransitions/ProgressTransitionManager.ts [hunk 3]
```diff
@@ -205,16 +205,21 @@ function createProgressTransitionRegister() {
 }
 
 if (shouldBeUseWeb()) {
-  const throwError = () => {
-    throw new Error(
-      '[Reanimated] ProgressTransitionRegister should not be used on web.'
-    );
+  const maybeThrowError = () => {
+    if (!isJest()) {
```
Comment (tjzel) [inline src/reanimated2/layoutReanimation/sharedTransitions/ProgressTransitionManager.ts:209]: Please add a comment about why do we have to if `jest` out here.

**Review by tjzel — CHANGES_REQUESTED** [review]

**Review by tjzel — APPROVED** [review]

**Comment by tjzel** [comment]
Please double check if web behavior is still proper.

**Review by piaskowyk — APPROVED** [review]

**Pull request merged**